### PR TITLE
Add AArch64 bit-field instructions (UBFX/SBFX/BFI/BFXIL/UBFIZ/SBFIZ)

### LINKED ISSUE #61: Add bitfield instructions (UBFX, SBFX, BFI, BFXIL, UBFIZ, SBFIZ)
## Motivation

Shift-and-mask patterns are extremely common (extracting fields, packing structs, fixed-point arithmetic) and currently require 2-3 instructions in this IR. AArch64 bit-field instructions collapse them to one. They are also a frequent target of compiler peepholes, so adding them lets the superoptimizer match or beat what `-O2` produces.

## Instructions

All take `{ rd, rn, lsb, width }` immediates where `lsb ∈ 0..=63`, `width ∈ 1..=64-lsb`:

- `UBFX rd, rn, #lsb, #width` — unsigned bit-field extract (zero-extended)
- `SBFX rd, rn, #lsb, #width` — signed bit-field extract (sign-extended)
- `UBFIZ rd, rn, #lsb, #width` — extract low `width` bits of `rn` and place at `lsb` in `rd` (zero rest)
- `SBFIZ rd, rn, #lsb, #width` — signed version of UBFIZ (sign-extends within the field)
- `BFI rd, rn, #lsb, #width` — insert low `width` bits of `rn` at `lsb` of `rd`, leaving other bits of `rd` unchanged
- `BFXIL rd, rn, #lsb, #width` — extract bits `[lsb+width-1:lsb]` of `rn` and place at `[width-1:0]` of `rd`, leaving high bits unchanged

(All are aliases of UBFM/SBFM/BFM; an implementation could expose UBFM/SBFM/BFM as the underlying IR variants and use the aliases only for display/encoding.)

## Scope

- IR variants (6, or 3 BFM forms + display aliases).
- Concrete semantics with bit-slice + (sign-)extend.
- SMT semantics: `extract`, `zero_extend`/`sign_extend`, `bvshl`, mask, and `bvor` for the insert variants.
- Encoder: UBFM/SBFM/BFM immediates use (immr, imms) — need helpers to convert from (lsb, width). Reject invalid (lsb,width) in `is_encodable_aarch64`.
- Cost model.

## Acceptance

- SMT proves `LSR t,a,#lsb; AND r,t,#((1<<width)-1)` ≡ `UBFX r,a,#lsb,#width` for representative (lsb,width).
- SMT proves a BFI sequence equivalent to `AND clear; AND mask; ORR`.
- Roundtrip via Capstone disassembly.

### PR DESCRIPTION
**Pull request by pmatos**
## Summary
- Adds the six 64-bit AArch64 bit-field aliases of UBFM/SBFM/BFM, end-to-end: IR + Display, dynasm encoder, parser, concrete + SMT semantics, cost model, is_encodable, enumerative + random candidate generation, and stochastic mutation cluster.
- `BFI` / `BFXIL` correctly report both `rd` and `rn` as source registers since they preserve unmodified bits of `rd` — distinct from the extract variants.
- 2D constraint `lsb + width <= 64` is enforced consistently in `is_encodable_aarch64`, parser, enumerative gen, random gen, and the mutation arm (which clamps `width` when `lsb` moves).
- Scope is 64-bit X-registers only; 32-bit W variants tracked separately (follow-up issue).

## Acceptance criteria (issue #61), both proved by SMT
- `UBFX rd,rn,#lsb,#width` ≡ `LSR t,rn,#lsb; AND rd,t,#((1<<width)-1)` — `test_ubfx_equivalent_to_lsr_and_mask`
- `BFI rd,rn,#lsb,#width` ≡ `AND/LSL/AND/ORR` expansion — `test_bfi_equivalent_to_and_and_or`
- Capstone roundtrip via existing `disassemble_and_verify` — `test_{ubfx,sbfx,bfi,bfxil,ubfiz,sbfiz}_correctness`

## Test plan
- [x] `./ci_check.sh` — fmt, build, unit tests, x86 test binaries, full suite (575 passing; +31 new since baseline 544).
- [x] `width = 64` boundary: `test_ubfx_full_width_is_identity` exercises the `1u64 << 64` UB guard.
- [x] `lsb = 63` upper boundary: `test_ubfx_high_lsb_boundary` (concrete) and `test_ubfx_high_lsb_correctness` (Capstone).
- [x] SP rejection in `rd` and `rn`: `test_is_encodable_bitfield`.
- [x] opcode_id drift guard between `candidate.rs` and `isa/aarch64.rs`: `test_bitfield_opcode_id_matches_isa_backend`.
- [x] Random generator must emit valid bit-field instructions: `test_generate_random_instruction_emits_bitfield_eventually` (5000 trials, every emission must pass `is_encodable_aarch64`).
- [x] Stochastic mutation produces a different, still-encodable bit-field within 200 trials, and reaches a peer variant via opcode mutation.

## Notes on alias collisions
Capstone's alias-decoder maps UBFX with `lsb + width == 64` (i.e., `imms == 63`) to `LSR` — they share the same UBFM bit pattern. The full-width correctness test only verifies encoding length; equivalence is covered by concrete + SMT.

Resolves #61.

### CODE REVIEW

**Comment by pmatos** [comment]
@codecov[bot] Regarding your coverage report ([comment](https://github.com/pmatos/s11/pull/143#issuecomment-4462898339)):

**Deferred (diminishing returns)** — The acceptance criteria from issue #61 (UBFX≡LSR+AND and BFI≡AND/AND/ORR) are both proved by direct SMT equivalence tests. The remaining four bit-field SMT lowering arms (SBFX, BFXIL, UBFIZ, SBFIZ) are exercised via concrete tests and via concrete↔SMT cross-check at the search level, but lack dedicated SMT-prove tests. That's a worthwhile coverage improvement and not a correctness blocker — adding four more SMT-prove tests is in-scope follow-up work.

Tracked as #146 (https://github.com/pmatos/s11/issues/146).

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). If you disagree, please reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@claude[bot] Regarding your review ([comment](https://github.com/pmatos/s11/pull/143#issuecomment-4462903283)), each sub-item has been triaged by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). Outcomes:

### B1 — "Missing regression test in `src/main.rs::convert_capstone_op_handles_all_supported_aarch64_mnemonics`"

**Not an issue** — Both evaluators verified that neither the named test nor the alleged CLAUDE.md requirement exists in this repo:

- `grep -rn "convert_capstone_op_handles_all_supported_aarch64_mnemonics" .` → no matches.
- `grep -n "regression test" CLAUDE.md` → no matches.

The cited "explicit CLAUDE.md requirement" appears to have been fabricated. No code change.

### B2 — `pick_non_sp` infinite loop in release builds

**Out of scope for this PR** — The hazard is real, but the identical pattern is pre-existing in the CCMP/CCMN block (`src/search/candidate.rs:488` on `main` before this PR). Fixing only the bit-field site would leave the same bug in the older code; fixing both is a wider refactor not specific to issue #61. Default register sets (CLI / search config) exclude SP, so the bug is unreachable in normal operation.

Tracked as #147 (https://github.com/pmatos/s11/issues/147).

### B3 — u8 arithmetic in `concrete.rs::Sbfx`

**Fixed in `3844c03`.**

Changed: widened `lsb`/`width` to `u32` before the sum at `src/semantics/concrete.rs:409` so unvalidated callers (future fuzz harnesses, raw API users) don't silently wrap through u8.
Validation: `./ci_check.sh` green (fmt, build, unit tests, x86 test binaries, full suite — 575 passing).

### B4 — Negative-path parser tests only cover UBFX

**Deferred (diminishing returns)** — All six bit-field dispatch arms route through the shared `parse_bfm_like` helper, so the helper's error branches are de facto tested by the existing `parse_ubfx_*` tests. Adding 5 near-duplicate positive roundtrip tests is good-to-have documentation but not a correctness blocker. The 31 new tests in this PR exercise each variant end-to-end via Display/encoder/concrete/SMT, so a dispatch-arm deletion wouldn't go unnoticed.

Tracked as #148 (https://github.com/pmatos/s11/issues/148).

### B5 — Stochastic `mutate_opcode` verbosity

**Style preference (no change)** — Pure stylistic. The current explicit-per-variant pattern matches how other peer-clusters in `src/search/stochastic/mutation.rs` are written (e.g., the CLZ/CLS/RBIT/REV cluster). A lookup-table refactor would be inconsistent with the surrounding code.

### B6 — Duplicate `opcode_id` table cross-reference comment

**Deferred** — The drift-guard test (`test_bitfield_opcode_id_matches_isa_backend`) already catches any divergence between the two `opcode_id` definitions and runs on every push. A one-line cross-reference comment near each duplicate is zero-risk documentation worth doing in a small follow-up, but not strictly necessary for this PR's correctness — and the observation generalises to every other instruction family in the two tables, so a comment scoped only to the bit-field additions would be inconsistent.

Tracked as #149 (https://github.com/pmatos/s11/issues/149).

---

_The verdicts above reflect a dual-evaluator triage (Claude Opus 4.7 and Codex GPT-5.x). If you disagree with any individual outcome, reply and we'll re-evaluate._

File: src/isa/aarch64.rs [hunk 1]
```diff
@@ -157,6 +157,12 @@ impl InstructionType for Instruction {
             Instruction::Umulh { .. } => 46,
             Instruction::Ccmp { .. } => 47,
             Instruction::Ccmn { .. } => 48,
+            Instruction::Ubfx { .. } => 49,
+            Instruction::Sbfx { .. } => 50,
+            Instruction::Bfi { .. } => 51,
+            Instruction::Bfxil { .. } => 52,
+            Instruction::Ubfiz { .. } => 53,
+            Instruction::Sbfiz { .. } => 54,
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/isa/aarch64.rs:172]: Fixed in `932c84e`.

Changed: bumped `AArch64InstructionInfo::opcode_count()` from 47 to 55 and extended `all_instruction_families()` to include CCMP/CCMN (IDs 47, 48) plus the six bit-field aliases (IDs 49–54). This pre-existing skew dated from the CCMP/CCMN merge — fixed all 8 families in one shot since the contract is `opcode_id() < opcode_count()`. Adding CCMP/CCMN to the family list surfaced a pre-existing test gap in `should_update_flags` (CCMP/CCMN write NZCV but weren't listed); fixed that too.

Validation: `./ci_check.sh` green — 575 unit tests pass (including the now-strengthened `all_instruction_families_cover_trait_methods` which iterates the full set of 55 families and asserts `opcode_id < opcode_count` for each).

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). If you disagree, please reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@claude[bot] Regarding your second review ([comment](https://github.com/pmatos/s11/pull/143#issuecomment-4463000610)) — this re-fires the same items as your first review with one new addition. Quick triage:

- **`src/main.rs::convert_capstone_op_handles_all_supported_aarch64_mnemonics`** — same as B1 in the first review. **Not an issue**: `grep -rn "convert_capstone_op_handles_all_supported_aarch64_mnemonics" .` returns no matches anywhere in the repo; CLAUDE.md contains no such requirement. The cited test does not exist. No change.

- **`pick_non_sp` infinite loop** — same as B2. Deferred to #147 (same hazard pre-existing in CCMP/CCMN, default register sets exclude SP).

- **Parser roundtrip tests only cover UBFX** — same as B4. Deferred to #148.

- **`parse_line_covers_all_core_mnemonics` may be stale** — **NEW** and correct: the list at `src/parser/mod.rs:1463` does not include the six new bit-field mnemonics. **Folded into #148** (its description already says "extend `parse_line_covers_all_core_mnemonics` to cover all six bit-field mnemonics"). No separate issue.

- **`opcode_count` was bumped to 55 in `932c84e`** in response to @chatgpt-codex-connector's separate inline review on `src/isa/aarch64.rs:165` — that fix supersedes any latent concern about the `opcode_id() < opcode_count()` contract.

_This triage matches the earlier dual-evaluator decision (Claude Opus 4.7 + Codex GPT-5.x). If you disagree with any individual outcome, reply and we'll re-evaluate._

**Comment by pmatos** [comment]
**Correction — B1 was actually correct, retrying.**

When I triaged @claude[bot]'s review I checked `CLAUDE.md` and `grep`ed for `convert_capstone_op_handles_all_supported_aarch64_mnemonics`, found no matches, and verdict-ed B1 as REJECT (`not-an-issue`). That was based on stale CLAUDE.md text on my local branch.

While this autofix loop was running, `main` advanced by 5 commits (b978395 → f3c6a0d). Commit `c62364b` ("Wire CCMP/CCMN into AArch64InstructionGenerator metadata") added both:

- The "Adding a new AArch64 instruction" section to `CLAUDE.md` (now lines 130–139), which explicitly mandates this test.
- The `convert_capstone_op_handles_all_supported_aarch64_mnemonics` test itself in `src/main.rs:1878`.

After `git pull --rebase origin main`, both are present locally. **B1 is therefore FIX, not REJECT** — fixed in `12d46e6`: added the six bit-field mnemonics (UBFX/SBFX/BFI/BFXIL/UBFIZ/SBFIZ) with canonical operand strings and bumped the tripwire row count from 52 to 58.

Branch was force-pushed after the rebase. PR is now based on `f3c6a0d`. Two earlier commits (`3844c03` Sbfx widening, `eabc95b` opcode_count bump) survived the rebase; `eabc95b` is the rebased-and-renamed version of `932c84e` with the merge conflict against `c62364b` resolved (taking the larger opcode_count=55 since the bit-field IDs go up to 54, and merging the doc comment).

Sorry for the confusion on B1 — the dual-evaluator triage was correct given the inputs I gave it (no matches in repo), but those inputs were a snapshot of a branch that had since gone stale. The other verdicts (REJECT B5, DEFER A/B2/B4/B6, FIX B3) stand.

File: src/isa/aarch64.rs [hunk 2]
```diff
@@ -1076,18 +1160,19 @@ impl InstructionGenerator<Instruction> for AArch64InstructionGenerator {
...
     /// `opcode_id()`). Not the same as `generate_random`'s slot count —
-    /// `generate_random` samples 27 top-level slots and folds ANDS, CSET,
-    /// CSETM, and ROR into a sub-multiplexer on slot 23 (plus the six
-    /// single-source bit-manipulation ops into a sub-multiplexer on slot 26)
-    /// to keep the slot table small. So `opcode_id < opcode_count` always
-    /// holds, but the random-generation distribution is not uniform across
-    /// all 42 IDs.
+    /// `generate_random` samples 28 top-level slots and folds several
+    /// families into sub-multiplexers (e.g. CLZ/CLS/RBIT/REV*/REV16 on
+    /// slot 26, and the six bit-field aliases on slot 28) to keep the slot
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/isa/aarch64.rs]: Fixed in `a17777c`.

Changed: bumped `AArch64InstructionGenerator::generate_random`'s sample range from `0..29` to `0..30` and added slot 29 as a 6-way sub-multiplexer for UBFX/SBFX/BFI/BFXIL/UBFIZ/SBFIZ. Uses the same SP-pre-filter + MNEG-fallback pattern as the existing CCMP slot (issue #57) to avoid the release-build infinite-spin hazard on degenerate `[SP]`-only register pools. 2D `lsb + width <= 64` constraint enforced by sampling width AFTER lsb so width is bounded by `64 - lsb` (matching `src/search/candidate.rs::generate_random_instruction` case 28).

Validation: `./ci_check.sh` green — 575 unit tests pass, including `test_random_instruction_generation` (which exercises this generator) and the strengthened `all_instruction_families_cover_trait_methods` that iterates the full 55-family set.

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). If you disagree, please reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@chatgpt-codex-connector[bot] Regarding your two Codex Review summary entries ([first](https://github.com/pmatos/s11/pull/143#pullrequestreview-4300951629) on `3cf8b7a` and [second](https://github.com/pmatos/s11/pull/143#pullrequestreview-4301037681) on `932c84e`):

**Already handled (no change to summaries themselves)** — both review summaries are the standard Codex Review wrapper ("Here are some automated review suggestions…") with no concrete ask in the summary body. The actionable feedback lives in the two attached inline review threads:

- *Raise opcode_count for bit-field opcodes* (`src/isa/aarch64.rs:165`) → **Fixed** in `eabc95b` (now part of the rebased PR head) — `opcode_count()` bumped to 55 and `all_instruction_families()` extended to cover CCMP/CCMN + all six bit-field aliases.
- *Add bit-field slot to ISA random generator* (`src/isa/aarch64.rs`) → **Fixed** in `a17777c` — `AArch64InstructionGenerator::generate_random` now samples `0..30` with a slot-29 6-way sub-multiplexer for the bit-field aliases.

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). If you disagree, please reply and we'll re-evaluate._

File: src/isa/aarch64.rs [hunk 3]
```diff
@@ -1169,20 +1316,21 @@ impl InstructionGenerator<Instruction> for AArch64InstructionGenerator {
...
-        49 // 20 original + 14 Tier 1 (MVN, NEG, NEGS, MovN, BIC, BICS, ORN,
+        55 // 20 original + 14 Tier 1 (MVN, NEG, NEGS, MovN, BIC, BICS, ORN,
         //  EON, ADDS, SUBS, ANDS, CSET, CSETM, ROR) + 2 MOVK/MOVZ (issue
         //  #55) + 6 single-source bit-manipulation (CLZ, CLS, RBIT, REV,
         //  REV32, REV16) + 5 multiply-accumulate family (issue #56:
-        //  MADD, MSUB, MNEG, SMULH, UMULH) + 2 conditional-compare
-        //  family (issue #57: CCMP, CCMN).
+        //  MADD, MSUB, MNEG, SMULH, UMULH) + 2 conditional-compare family
+        //  (issue #57: CCMP, CCMN) + 6 bit-field aliases (UBFX, SBFX, BFI,
+        //  BFXIL, UBFIZ, SBFIZ, issue #61).
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/isa/aarch64.rs:1401]: Fixed in `97b014a`.

Changed: extended `AArch64InstructionGenerator::generate_all` to enumerate the six bit-field families with sparse `(lsb, width)` sampling (5 lsb × 6 width × 6 variants × 31² non-SP register pairs ≈ 138k extra candidates) — mirrors the sampling tables already in `src/search/candidate.rs::generate_all_instructions`. SP is pre-filtered from both rd and rn (matches `is_encodable_aarch64`), and the 2D constraint `lsb + width <= 64` is enforced inline.

Validation: `./ci_check.sh` green — 575 unit tests still pass, including `generate_all_covers_every_aarch64_family` which iterates all 55 opcode IDs and asserts ID coverage from the generated set.

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). If you disagree, please reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@claude[bot] Regarding your two further reviews ([3rd review](https://github.com/pmatos/s11/pull/143#issuecomment-4463183520) on `a17777c` and [4th review](https://github.com/pmatos/s11/pull/143#issuecomment-4463229616) on `a17777c`), batched triage:

### From the 3rd review

- **`debug_assert!` on SBFX/SBFIZ shift_left invariant** — **No action taken**. The invariant `lsb + width ≤ 64 ∧ width ≥ 1 ⇒ shift_left ∈ [0..63]` is enforced upstream by `is_encodable_aarch64`. Adding a redundant `debug_assert!` at the apply-instruction site duplicates that check on every concrete-interpreter call. The widening commit (`3844c03`) already addressed the only realistic unvalidated-caller hazard.
- **`pick_non_sp` infinite loop** — **Already tracked** in #147 (covers all `pick_non_sp` sites and the broader SP-filtering pattern).
- **Redundant `.max(1)` in mutation.rs** — **Style preference (no change)**. The guard is harmless. The masked `lsb` is bounded by `0x3F` so `64 - lsb ≥ 1` is structurally guaranteed; removing `.max(1)` is correct but the explicit floor documents the post-condition for readers who haven't traced the mask. Net zero behavioural change either way.
- **Multiply-accumulate weight doubles silently** — **Out of scope for this PR**. Real observation: bumping `random_range(0..30)` to `0..31` made the `_` arm catch both 29 and 30, doubling MNEG/SMULH/UMULH frequency. This is a sampling-distribution concern that touches the existing MADD/MSUB family rather than the bit-field addition, and the same applies every time a new slot is added. Will track as a separate audit of the slot-distribution model when it next matters. The bit-field family did get its own slot (28) so its weight is correct.
- **`mutate_opcode` cluster verbosity** — same as B5 in the first review. **Style preference (no change)**.
- **Missing SBFIZ boundary test at `lsb=0, width=64`** — **Deferred (diminishing returns)**, folded into the SMT-coverage follow-up #146. The existing `test_sbfiz_sign_extends_field_above_lsb_plus_width` and `test_sbfiz_positive_field` exercise the sign-extension path; a `lsb=0, width=64` SBFIZ degenerates to "rd ← rn" which is also covered indirectly by `test_ubfx_full_width_is_identity` at the SMT level (UBFX with the same `(0, 64)` is the identity).

### From the 4th review

- **SP leak in `mutate_operand` for bit-field rd/rn** — **Deferred**, folded into #147. `Mutator::random_register` does not filter SP, but `default_mutator()` and the production register pools (`src/main.rs`, `src/search/config.rs`) never include SP, so the test `test_bitfield_operand_mutation_changes_fields_and_stays_encodable` passes. Same latent hazard family as the `pick_non_sp` loop concern.
- **Stale `29 top-level slots` in `opcode_count` docstring** — **Fixed in `97b014a`**. Corrected to `30 top-level slots`.
- **SMT equivalence test for SBFX** — **Already tracked** in #146.
- **`width_raw` signed comparison in `parse_bfm_like`** — **Style preference (no change)**. `parse_immediate` returns `i64`; `(1..=64).contains(&width_raw)` correctly rejects negative `width_raw` (negatives aren't in the range). Adding an explicit `if width_raw < 0` guard would be redundant.
- **`SBFX` defensive comment misleading** — **Style preference (no change)**. The comment is precise: it documents what the widening protects against ("unvalidated callers"), which is the design rationale for the change. Even though `is_encodable_aarch64` is the only enforced entry point today, a future fuzz harness or raw-IR test calling `apply_instruction_concrete` directly would bypass it.

_The verdicts above reflect dual-evaluator triage (Claude Opus 4.7 and Codex GPT-5.x). If you disagree with any individual outcome, reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@chatgpt-codex-connector[bot] Regarding your latest Codex Review summary ([on `a17777c400`](https://github.com/pmatos/s11/pull/143#pullrequestreview-4301133450)) and your conversational note ([env setup link](https://github.com/pmatos/s11/pull/143#issuecomment-4463215791)):

- **Codex Review summary** (`a17777c400`) — **Already handled (no change)**: standard wrapper boilerplate; the actionable inline thread `Add bit-fields to generate_all` it attaches to has been addressed — **fixed in `97b014a`**, `AArch64InstructionGenerator::generate_all` now enumerates the six bit-field families with the same sparse `(lsb, width)` table used in `src/search/candidate.rs`.
- **"To use Codex here, create an environment for this repo"** — **Not applicable**: infrastructure setup note for the repository, not an actionable PR request.

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). If you disagree, please reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@claude[bot] Regarding your [5th review](https://github.com/pmatos/s11/pull/143#issuecomment-4463356716) on `97b014a`:

- **Distribution skew in `candidate.rs`** — **Out of scope for this PR**. Real observation (bit-field at slot 28 has 1/31 weight while multiply-accumulate via the `_` catch-all catches 2 of 31 values → ~2x weight) but the asymmetry pre-dates this PR — multiply-accumulate has had 2x weight relative to other explicit slots since it was added in `dd9a21c`. Equalising slot weights is a clean refactor (move multiply-accumulate to its own explicit slot, change `_ =>` to `_ => unreachable!()`) but it touches every family's sampling weight slightly and deserves its own review pass. The sister generator `isa/aarch64.rs::generate_random` already uses the cleaner explicit-slot + `unreachable!()` pattern; bringing `candidate.rs` in line is the right follow-up. Note: your framing "multiply-accumulate probability doubled" is technically slightly off — its absolute weight went from 2/30 to 2/31 (a tiny decrease); what's true is that bit-field is 1x while multiply-accumulate is 2x. Tracked as #153 (https://github.com/pmatos/s11/issues/153).
- **`pick_non_sp` infinite loop in release builds** — **Already tracked** in #147 (covers all `pick_non_sp` sites and the broader SP-filtering pattern).
- **SBFX with `width = 64` concrete test gap** — **Already tracked** in #146 (SMT-equivalence coverage for the non-UBFX/BFI variants, including boundary cases).
- **`parse_ubfx_rejects_*` negative tests only on one mnemonic** — **Already tracked** in #148 (per-variant parser roundtrip tests).
- **Slot-number cross-reference comment** (`candidate.rs` uses 28, `isa/aarch64.rs` uses 29) — **Style preference (no change)**. The two numberings are intentionally independent and documented in `opcode_count()`'s docstring on `isa/aarch64.rs` ("Not the same as `generate_random`'s slot count"). Adding cross-reference comments in `candidate.rs` is good documentation hygiene but isn't load-bearing for correctness.
- **Correctness highlights** (UB guard, SBFX shift soundness, BFI/BFXIL source-register handling, Capstone alias-collision documentation, opcode_id drift guard, 2D constraint consistency) — **No action**: thank you for the verification.

_The verdicts above reflect dual-evaluator triage (Claude Opus 4.7 and Codex GPT-5.x). If you disagree with any individual outcome, reply and we'll re-evaluate._

**Comment by pmatos** [comment]
@chatgpt-codex-connector[bot] Regarding your [refreshed env-setup note](https://github.com/pmatos/s11/pull/143#issuecomment-4463293709) — **Not applicable (no change)**: same boilerplate as your earlier note ([id 4463215791](https://github.com/pmatos/s11/pull/143#issuecomment-4463215791)). Repository infrastructure setup hint, not an actionable PR request. No action.

_This assessment was made by two independent AI reviewers (Claude Opus 4.7 and Codex GPT-5.x). If you disagree, please reply and we'll re-evaluate._
